src/jacobians.jl

```diff
@@ -124,6 +124,62 @@ function finite_difference_jacobian(f, x::AbstractArray{<:Number},
     finite_difference_jacobian(f, x, cache, f_in; relstep=relstep, absstep=absstep, colorvec=colorvec, sparsity=sparsity, dir=dir)
 end
 
+@inline function _colorediteration!(J,sparsity,rows_index,cols_index,vfx,colorvec,color_i,ncols)
+    @inbounds for i in 1:length(cols_index)
+        if colorvec[cols_index[i]] == color_i
+            J[rows_index[i],cols_index[i]] = vfx[rows_index[i]]
+        end
+    end
+end
+
+@inline function _colorediteration!(J,sparsity::SparseMatrixCSC,rows_index,cols_index,vfx,colorvec,color_i,ncols)
+    @inbounds for col_index in 1:ncols
+        if colorvec[col_index] == color_i
+            @inbounds for row_index in view(sparsity.rowval,sparsity.colptr[col_index]:sparsity.colptr[col_index+1]-1)
+                J[row_index,col_index]=vfx[row_index]
+            end
+        end
+    end
+end
+
+#override default setting of using findstructralnz
+_use_findstructralnz(sparsity) = ArrayInterface.has_sparsestruct(sparsity)
+_use_findstructralnz(::SparseMatrixCSC) = false
+
+function __init__()
+    @require BlockBandedMatrices="ffab5731-97b5-5995-9138-79e8c1846df0" begin
+        _use_findstructralnz(::BlockBandedMatrices.BandedBlockBandedMatrix) = false
+
+        @inline function _colorediteration!(Jac::BlockBandedMatrices.BandedBlockBandedMatrix,
```

Comment:
What about just standard BlockBandedMatrix and BandedMatrix? Are those fine without a special iteration?

Reply:
Yeah, I'm working on that.